### PR TITLE
Crane mappings

### PR DESCRIPTION
**Pull request by pdalpra**
This will be required by the upcoming `trigger-functional-tests` enhancement (eg. not launching every single IT on merge to master), as we cannot rely on tool-stack-launcher (which now defers to `teads-central crane env`) to find the components to test.

Instead, we'll rely on `teads-central crane mapping master --from-repository <repository>`, which will list the components built by `<repository>` (passed as an additional non mandatory parameter to `trigger-functional-tests`) and rely on `crane status <image>+affected` to find the effective list of components affected by that merge, so that we'll still launch the same subset of ITs  that PRs builds does.

### CODE REVIEW

**Comment by pdalpra** [comment]
cc @ebuzzing/ci-cd

File: docker/crane-mappings.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
+package docker
```
Comment (bjaglin) [inline docker/crane-mappings.go]: I have the feeling that it's an anti-pattern to have hyphens in a file name... maybe it's time to create a crane subpackage?

File: cmd/crane/mappings.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+var mappingsCmd = &cobra.Command{
+	Use:   "mappings",
+	Short: "Extract Github <-> Docker mapping from crane-config",
+	Long: `mappings extracts the Github <-> Docker mappings from crane-config image.
+To find from which Github repository a Docker image originates, specify the image name using the --from-image flag.
+To find which images originates from a Github repository, specify the repository name using the --from-repository  flag.
+
+The crane-config tag to search mappings in can be specified by passing the branch as the first argument (defaults to 'master').
+
+For example, if working on a feature with a git branch name my_new_component, you can:
```
Comment (bjaglin) [inline cmd/crane/mappings.go]: does it imply that a branch of that name exists for `crane-config`? does it fallback? (I could check the code but I am trying to stick to the doc for now)

File: cmd/crane/mappings.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	- 'teads-central crane mappings my_new-component --from-repository myrepo' will lookup Docker images originating from the 'myrepo' Github repository
+	- 'teads-central crane mappings my_new-component --from-image myservice' will lookup the Github repository from which the 'myservice' Docker image originates`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var (
+			mappings    []string
+			craneBranch = environment.DockerNameFriendly(craneConfigBranch(args))
```
Comment (bjaglin) [inline cmd/crane/mappings.go]: if it's a branch (which is what the var name seems to indicate), it shouldn't go through `DockerNameFriendly`

File: docker/crane-mappings_test.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+import (
+	"bytes"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestCraneMappingsFromRepository(t *testing.T) {
+	Convey("Given the tag on crane-config does not exist", t, func() {
```
Comment (bjaglin) [inline docker/crane-mappings_test.go]: see my comment about the doc, shouldn't we fallback? that's at least what we expect for the feature we want, but the "retry on master" logic could very well be implemented outside teads-central

File: docker/crane-mappings.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	impl "github.com/fsouza/go-dockerclient"
+	"gopkg.in/yaml.v2"
+)
+
+const craneYmlPath = "/crane.yml"
```
Comment (bjaglin) [inline docker/crane-mappings.go]: I'd add a comment to show where this comes from - https://github.com/ebuzzing/tool-crane-config/blob/a2782720d5930b1421be7d63c998986baad96c36/Dockerfile#L147

**Comment by bjaglin** [comment]
Great work!

Two notes about the future usage of it (so not directly connected to this PR, but related):
* We cannot rely on `+affected` as it only works for started containers, you should use `+dependencies` in the reverse order, see the ugly https://github.com/ebuzzing/scripts-stack-helpers/blob/master/affects.sh
* Since the mapping will become so critical, I think we should add a sanity check when retrieving the mappings in trigger-functional-tests, to verify that the affected components are a superset of the images retrieved from the mapping (and otherwise fail-fast, to detect a typo/something missing in the `git:` fields, or a problem in the tooling)

File: docker/crane-mappings.go (hunk 1 above)
Comment (pdalpra) [inline docker/crane-mappings.go]: Do you mind if I get this in another PR ?
There's quite a few things that are getting too messy for my own taste in teads-central, I'm planning a refactoring :)

File: cmd/crane/mappings.go (hunk 2 above)
Comment (pdalpra) [inline cmd/crane/mappings.go]: If the branch is specified, yes, the corresponding tag must exist for crane-config.

Comment (pdalpra) [inline cmd/crane/mappings.go]: `master` is used only if the branch is not specified at all.

File: cmd/crane/mappings.go (hunk 3 above)
Comment (pdalpra) [inline cmd/crane/mappings.go]: woops, forgot to rename it...

File: docker/crane-mappings_test.go (hunk 4 above)
Comment (pdalpra) [inline docker/crane-mappings_test.go]: In fact, for the feature we want, we'll be using teads-central if the build parameter specifying the repo were a PR was just merged is set on Jenkins, and only `_master` builds will set it, therefore we won't even specify the branch.
The possibility to choose another branch is mostly there for usage of `teads-central crane  mappings` outside `trigger-functional-tests`, eg. by developers.

**Comment by pdalpra** [comment]
added a comment to link to the place where `/crane.yml` is set in the Dockerfile + renamed `craneBranch` to `craneTag` in mappings.go

File: docker/crane-mappings.go (hunk 1 above)
Comment (bjaglin) [inline docker/crane-mappings.go]: you're right, better do that in a no-op PR

**Pull request merged**